packages/rspeedy/plugin-react-alias/test/compat/legacy.test.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import { createRsbuild } from '@rsbuild/core'
+import { describe, expect, test, vi } from 'vitest'
+
+import { LAYERS } from '@lynx-js/react-webpack-plugin'
+
+vi.mock('node:fs', async (importOriginal) => {
+  const original = await importOriginal()
+  const { readFileSync: originalReadFileSync } =
+    original as typeof import('node:fs')
+  return {
+    ...original as object,
+    readFileSync: vi.fn().mockImplementation((filePath: string) => {
+      if (
+        /[\\/](?:packages[\\/])?react[\\/]*package\.json$/.test(filePath)
+      ) {
+        return JSON.stringify({ version: '0.111.0' })
+      } else {
+        return originalReadFileSync(filePath)
+      }
+    }),
+  }
+})
+
+describe('@lynx-js/react/compat - alias', () => {
+  test('alias with @lynx-js/react < 0.112.0', async () => {
```

Comment:
Can we also add some tests for canary version?

E.g.:

- `@lynx-js/react@0.111.999-canary-20250728-1f7b2d07`
- `@lynx-js/react@0.112.0-canary-20250728-1f7b2d07`
- `@lynx-js/react@0.112.1-canary-20250730-1f7b2d07`